C-Plus-Plus/cp/Peak_element.cpp

```diff
@@ -0,0 +1,127 @@
+/** An array element is a peak if it is NOT smaller than
+its neighbours. For corner elements, we need to
+consider only one neighbour.
+Since an array always have a maximum value therefore an peak element always be there and there can be many peak element
+**/
+
+#include <bits/stdc++.h>
+using namespace std;
+
+// recursive approach
+int find_peak_recursive(vector<int>v,int l,int r)
+{
+    if(l<r)
+    {
+        int mid = l+(r-l)/2;
+	// for corner cases
+        if(mid == 0 || mid+1 == v.size()) 
+        {
+            if(mid==0)
+            {
+                if(v[0] >= v[1])
+                return v[0];
+                else
+                return v[1];
+            }
+            else{
+                if(v[mid] >= v[mid-1])
+                return v[mid];
+                else
+                return v[mid-1];
+            }
+        }
+        
+        else if(v[mid]>=v[mid-1] && v[mid]>=v[mid+1])
+        return v[mid];
+        
+        else if(v[mid]<v[mid+1])
+        return find_peak_recursive(v,mid+1,r);
+        
+        else 
+        return find_peak_recursive(v,l,r-1);
+    }
+    return -1;
+}
+
+// using while loop approach
+int find_peak_iter(vector<int>arr)
+{
+    
+    int start=0,n=arr.size();
+    int end=n-1;
+    while(start<=end)
+    {
+        int mid=start+(end-start)/2;
+        if(mid-1>=0&&mid+1<n)
+        {
+            if(arr[mid]>arr[mid-1]&&arr[mid]>arr[mid+1])
+            {
+                return arr[mid];
+            }
+            else if(arr[mid]<arr[mid-1])
+            {
+                end=mid-1;
+            }
+            else
+            {
+                start=mid+1;
+            }
+        }
+	// for corner cases
+        else if(mid==0)
+        {
+            if(arr[0]>arr[1])
+            {
+                return arr[0];
+            }
+            else
+            {
+                return arr[1];
+            }
+        }
+	// for corner cases
+        else if(mid==n-1)
+        {
+            if(arr[n-1]>arr[n-2])
+            {
+                return arr[n-1];
+            }
+            else
+            {
+                return arr[n-2];
+            }
+        }
+    }
+    return -1;
+}
+
+int main() {
+	int n;
+	cin>>n;
+	vector<int> v(n);
+	for(int i=0;i<n;i++)
+	cin>>v[i];
+	
+	int peak_recursive = find_peak_recursive(v,0,n-1);
+	cout << "Peak element ( Recursive approach ) : "<< peak_recursive <<"\n";
+	
+	int peak_iter = find_peak_iter(v);
+	cout << "Peak element ( Iterative approach ) : "<< peak_iter <<"\n";
+	
+	return 0;
+}
+
+/** 
+Eg. :
+Input: array[] = {10, 20, 15, 2, 23, 90, 67}
```

Comment:
Your sample input should replicate the input you will give when the program is run in console or any IDE.